ktlint-ruleset-standard/src/test/kotlin/com/github/shyiko/ktlint/ruleset/standard/ImportOrderingRuleTest.kt

```diff
@@ -21,4 +24,94 @@ class ImportOrderingRuleTest {
             )
         ).isEmpty()
     }
+
+    private val formattedImports =
+        """
+        import android.app.Activity
+        import android.view.View
+        import android.view.ViewGroup
+        import java.util.List
+        import kotlin.concurrent.Thread
+        """.trimIndent()
+
+    private val errorMessage = "Imports must be ordered in lexicographic order in a single group"
+
+    @Test
+    fun testFormatOk() {
+        assertThat(ImportOrderingRule().lint(formattedImports)).isEmpty()
+        assertThat(ImportOrderingRule().format(formattedImports)).isEqualTo(formattedImports)
+    }
+
+    @Test
+    fun testFormatWrongOrder() {
+        val imports =
+            """
+            import android.view.ViewGroup
+            import android.view.View
+            import android.app.Activity
+            import kotlin.concurrent.Thread
+            import java.util.List
+            """.trimIndent()
+        assertThat(ImportOrderingRule().lint(imports)).isEqualTo(
+            listOf(LintError(1, 1, "import-ordering", errorMessage))
+        )
+        assertThat(ImportOrderingRule().format(imports)).isEqualTo(formattedImports)
+    }
+
+    @Test
+    fun testFormatWrongOrderAndBlankLines() {
+        val imports =
+            """
+            import android.view.ViewGroup
+
+
+            import android.view.View
+            import android.app.Activity
+
+            import kotlin.concurrent.Thread
+
+            import java.util.List
+            """.trimIndent()
+        assertThat(ImportOrderingRule().lint(imports)).isEqualTo(
+            listOf(LintError(1, 1, "import-ordering", errorMessage))
+        )
+        assertThat(ImportOrderingRule().format(imports)).isEqualTo(formattedImports)
+    }
+
+    @Test
+    fun testFormatBlankLinesForMajorGroups() {
```

Comment:
redundant with `testFormatBlankLines`?

Reply:
It is, I think I did it for explicitness that major groupings will be removed, but it's been some time since I wrote it. I can remove it if you want.

Reply:
Removed it 918bda59f0d730aa927b00569f3d15a4f2ba9384